inc/namespace.php

```diff
@@ -81,14 +77,19 @@ function generate_docs_for_module( $id, Module $module ) : ?Group {
 	$group = new Group( $module->get_title() );
 
 	// Generate objects for each file.
-	$iterator = new RecursiveIteratorIterator(
-		new RecursiveDirectoryIterator(
-			$doc_dir,
-			ITERATOR_FLAGS
-		)
-	);
+	$iterator = new DirectoryIterator( $doc_dir );
 	foreach ( $iterator as $leaf ) {
 		/** @var \SplFileInfo $leaf */
+		if ( $leaf->isDir() ) {
+			if ( $leaf->isDot() ) {
+				continue;
+			}
+			// Special handling for sub dirs, to add a page (and sub pages).
+			$doc = get_page_for_dir( $leaf->getPathname(), $doc_dir );
```

Comment:
What if there's no readme? Or sub-sub dirs? Seems to me like a better way to handle this would be to stick to the RecursiveDirectoryIterator and work that way. We may want to replace the existing objects entirely there too.

Reply:
The dir is skipped if there's no readme, which I think we probably want, as we need a label for the sidebar too. Because we are storing the pages hierachically, it's better to allow the call path to be recursive too.

Reply:
Ah, it's handled inside `get_page_for_dir`, I see. 👍